### PR TITLE
Update request money header title

### PR DESCRIPTION
**Pull request by DylanDylann**
### Details

1. App displays 'Split' in title on back from scan request money
2. App displays 'Split' in title on back from manual request money

### Fixed Issues

$ https://github.com/Expensify/App/issues/28751
PROPOSAL: https://github.com/Expensify/App/issues/28751#issuecomment-1746280710

### Tests

Part 1:
1. Open the app
2. Click on plus and click request money->scan
3. Select any file
4. Observe now it displays 'Scan' in header, select any user
5. Click on back and verify that now it displays 'Scan' in header

Part 2:
1. Click manual tap
2. Enter amount and select any user
3. Click back button and verify that header message is 'Manual'
4. Click split button and verify that header message is changed to 'Split'

- [x] Verify that no errors appear in the JS console

### QA Steps

Part 1: 
1. Open the app
2. Click on plus and click request money->scan
3. Select any file
4. Observe now it displays 'Scan' in header, select any user
5. Click on back and verify that now it displays 'Scan' in header

Part 2:
1. Click manual tap
2. Enter amount and select any user
3. Click back button and verify that header message is 'Manual'
4. Click split button and verify that header message is changed to 'Split'

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] If we are not using the full Onyx data that we loaded, I've added the proper selector in order to ensure the component only re-renders when the data it is using changes
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/141406735/a3c71625-c056-4a1c-af10-89ec82390060

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/141406735/785b7bf2-858a-45d3-9b5e-42a0d4d29c83

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/141406735/a42afd67-5355-492f-819a-e2efe95dd40d

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/141406735/4110ec79-05f6-46c1-b44f-dc24525abe44

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/141406735/9bfcfc23-d516-4b9c-84ca-ab055a849645
</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/141406735/69c9fb63-0c6d-459f-a77f-dc83399991c6

</details>

### CODE REVIEW

**Comment by DylanDylann** [comment]
@fedirjh Could you help to take a look at this PR? Thanks

**Comment by DylanDylann** [comment]
@fedirjh Could you help to check this comment? https://github.com/Expensify/App/pull/29936#discussion_r1368774468

**Comment by DylanDylann** [comment]
@fedirjh I am trying your suggestion, i will give some response asap

**Comment by DylanDylann** [comment]
@fedirjh I just updated code according to your logic, please check again

**Review by fedirjh — COMMENTED** [review]
Let's simplify the code a bit.

File: src/libs/actions/IOU.js [hunk 1]
```diff
@@ -2834,6 +2834,13 @@ function setMoneyRequestParticipants(participants) {
     Onyx.merge(ONYXKEYS.IOU, {participants});
 }
 
+/**
+ * @param {Boolean} isSplitRequest
+ */
+function setMoneyRequestIsSplitRequest(isSplitRequest) {
+    Onyx.merge(ONYXKEYS.IOU, {isSplitRequest});
+}
```
[the remark this answers is not in the document]
Comment (DylanDylann) [reply src/libs/actions/IOU.js]: @fedirjh Thank you for your comment, please check again

File: src/libs/actions/IOU.js [hunk 2]
```diff
@@ -2829,9 +2829,11 @@ function setMoneyRequestBillable(billable) {
 
 /**
  * @param {Object[]} participants
+ * @param {Boolean} isSplitRequest
  */
-function setMoneyRequestParticipants(participants) {
+function setMoneyRequestParticipants(participants, isSplitRequest) {
     Onyx.merge(ONYXKEYS.IOU, {participants});
+    Onyx.merge(ONYXKEYS.IOU, {isSplitRequest});
```
[the remark this answers is not in the document]
Comment (DylanDylann) [reply src/libs/actions/IOU.js]: @fedirjh Quickly updated

**Comment by DylanDylann** [comment]
> @DylanDylann We should reset the `isSplitRequest` to `false` inside `IOU.resetMoneyRequestInfo` to fix [this issue](https://github.com/Expensify/App/pull/29936#discussion_r1372170952).

Thanks for your suggestion, I updated

File: src/libs/actions/IOU.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (DylanDylann) [reply src/libs/actions/IOU.js]: > cc @DylanDylann I think there is one more change needed, let's pass an empty array when `iou.isSplitRequest` is `false`.

@fedirjh  I think it is unnecessary, currently as code change we have 2 cases `iou.isSplitRequest = false`
1. In function `addSingleParticipant` (manual case) we must pass participants param
2. In function `addParticipantToSelection`, `isSplitRequest = false` when `newSelectedOptions.length === 0` which means `newSelectedOptions` is empty

**Review by fedirjh — CHANGES_REQUESTED** [review]
@DylanDylann Please address this bug, https://github.com/Expensify/App/pull/29936#discussion_r1373001768

**Comment by DylanDylann** [comment]
@fedirjh Thanks for your suggestion, I updated. Please check again

**Review by fedirjh — APPROVED** [review]
Looks good to me.

**Review by amyevans — APPROVED** [review]

**Pull request merged**